core/src/mindustry/ui/fragments/ChatFragment.java

```diff
@@ -216,14 +222,35 @@ public void hide(){
     }
 
     public void updateChat(){
-        chatfield.setText(history.get(historyPos));
-        chatfield.setCursorPosition(chatfield.getText().length());
+        chatfield.setText(mode.normalizedPrefix() + history.get(historyPos));
+        updateCursor();
+    }
+
+    public void nextMode(){
+        ChatMode prev = mode;
+
+        do{
+            mode = mode.next();
+        }while(!mode.isValid());
+
+        try{
+            chatfield.setText(mode.normalizedPrefix() + chatfield.getText().substring(prev.normalizedPrefix().length()));
+        }catch(IndexOutOfBoundsException e){ // in case the player removed a prefix
```

Comment:
Catching exceptions like these is just bad style; you should validate it instead.

Reply:
Resolved.